$NetBSD: patch-meson.build,v 1.4 2025/03/12 09:54:41 hauke Exp $

Make PREFIX path configurable.

Quota support on NetBSD is currently broken, pull up an upstream
change to disable it explicitly.

While NetBSD has libwrap, the meson code doesn't find it. The test
cond still succeeds, so add a branch that does not require the
library.

Install at.h on Linux, we do it on NetBSD, too.

--- meson.build.orig	2025-02-10 20:41:20.000000000 +0000
+++ meson.build
@@ -170,7 +170,7 @@ elif host_os == 'netbsd'
         'include',
         'sys',
         'etc/afpd',
-        '/usr/pkg/include',
+        get_option('prefix') / 'include',
     )
 elif host_os == 'omnios'
     root_includes = include_directories(
@@ -219,7 +219,7 @@ if host_os in ['dragonfly', 'freebsd', '
 endif
 
 if host_os == 'netbsd'
-    libsearch_dirs += '/usr/pkg/lib'
+    libsearch_dirs += get_option('prefix') / 'lib'
 endif
 
 if uname.found() and uname_stdout.to_lower().contains('omnios')
@@ -1025,6 +1025,12 @@ endif
 
 enable_quota = get_option('with-quota')
 
+if host_os == 'netbsd'
+    enable_quota = false
+    cdata.set('NO_QUOTA_SUPPORT', 1)
+    warning('quota is broken on NetBSD, see https://github.com/Netatalk/netatalk/issues/1225')
+endif
+
 prop = cc.find_library('prop', required: false)
 quota = cc.find_library('quota', required: false)
 rpcsvc = cc.find_library('rpcsvc', required: false)
@@ -2193,7 +2199,6 @@ if host_os.contains('freebsd')
     cdata.set('FREEBSD', 1)
     cdata.set('OPEN_NOFOLLOW_ERRNO', 'EMLINK')
 elif host_os.contains('linux')
-    use_glibc_at_header = cc.has_header('netatalk/at.h')
     cdata.set('_GNU_SOURCE', 1)
     if cc.compiles(
         '''
